src/query/interpret/eval.hpp

```diff
@@ -534,7 +630,22 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
 
   TypedValue Visit(MapProjectionLiteral &literal) override {
     TypedValue::TMap result(ctx_->memory);
-    for (const auto &pair : literal.elements_) result.emplace(pair.first.name, pair.second->Accept(*this));
+    for (const auto &pair : literal.elements_) {
+      if (pair.first.name == "*") {  // AllPropertiesSelector
+        auto maybe_selector = pair.second->Accept(*this);
+
+        if (maybe_selector.type() != TypedValue::Type::Map) {
+          throw QueryRuntimeException("Expected a map from all properties selection, got {}.", maybe_selector.type());
+        }
+
+        for (const auto &pair : maybe_selector.ValueMap()) {
```

Comment:
I would suggest unwrapping pair into two variable in both inner and outer loop because here you have variable overloading which makes it hard to read

Reply:
Also, I think you can change this inner for loop with `map::merge`. It will perform moving of elements from source to target map